CompanyApi/Controllers/CompanyController.cs

```diff
@@ -20,6 +23,75 @@ public ActionResult<Company> Create(CreateCompanyRequest request)
             return StatusCode(StatusCodes.Status201Created, companyCreated);
         }
 
+        [HttpGet]
+        public ActionResult<List<Company>> GetAllCompanies([FromQuery] int? pageSize, [FromQuery] int? pageIndex)
+        {
+            if (pageIndex == null || pageSize == null)
+            {
+                return StatusCode(StatusCodes.Status200OK, companies);
+            }
+
+            if ((pageIndex * pageSize) > companies.Count())
+            {
+                return StatusCode(StatusCodes.Status204NoContent);
+            }
+            var ressultCompanies = companies.Skip((int)((pageIndex - 1) * pageSize)).Take((int)pageSize).ToList();

```

Comment:
well: nice to use Skip Take
sugesstion: extract (pageIndex-1) * pageSize to a temp variable